src/backend/distributed/commands/table.c

```diff
@@ -701,13 +701,81 @@ PostprocessAlterTableSchemaStmt(Node *node, const char *queryString)
 }
 
 
+static char *
+GenerateIndexConstraintName(const char *tabname, Oid namespaceId, Constraint *constraint)
+{
+	char *conname = NULL;
+
```

Comment:
You are using ChooseIndexName in all the subcases of this function. Maybe here you can declare all the variables with meaningful names, e.g. something like:
bool primary
bool isconstraint
List *indexParams
List *excludeOpNames

And then set these values in each subcase: it also makes the difference in generating the name between the subcases clearer